spring-integration-ip/src/main/java/org/springframework/integration/ip/udp/UnicastSendingMessageHandler.java

```diff
@@ -290,6 +311,9 @@ protected DatagramSocket getTheSocket() {
 	}
 
 	protected synchronized DatagramSocket getSocket() throws IOException {
+		if (this.socketExpression != null) {
+			return this.socketExpression.getValue(this.evaluationContext, DatagramSocket.class);
```

Comment:
I think we are missing here the main gain from `Expression` stuff. The context to evaluate.
I just mean the `requestMessage` as a root object.
So, this `getSocket()` logic with the `socketExpression` should be moved to the `handleMessageInternal()`.